### PR TITLE
Fixed build failures from Github API Throttling Limits

### PR DESCRIPTION
**Pull request by dcwangmit01**
* Require that the user provide a GITHUB_API_TOKEN that is
  used upon calling the Github API to figure out latest releases.
* This fixes random CircleCI build failures, since non-auth'd
  requests are counted by egress IP address.
* This also fixes local build failures.
* User now required to set env var GITHUB_API_TOKEN

### CODE REVIEW

**Comment by dcwangmit01** [comment]
https://github.com/cisco-sso/kdk/issues/180

**Comment by hangxie** [comment]
Like CI reported, we will have to leave git token in this repo for this change but that's not something good ...

**Comment by dcwangmit01** [comment]
@rtluckie Prior PRs changed the build in a way that requires about 30 calls to the Github API in order to find the latest released versions of software.  Doing so broke things because public calls to the GithubAPI are very thottled.  We need to get beyond throttling limits by using a GITHUB_TOKEN. But, Travis doesn’t allow PRs from Forks to access Secrets. So now we're in a catch 22. The Fork PR will probably never build right.

Basically, I think we’re forced to move CI to Github CI. I bet there aren’t the API limits from their system.  @rtluckie Since you created the dependency on GithubAPI, do you want to fix it by moving us to Github CI?

**Comment by dcwangmit01** [comment]
I'm going to merge this.  The mainline CircleCI build should work, since it should have access to the GITHUB_API_TOKEN secret embedded in the .travis.yaml
